src/mono/mono/mini/interp/interp-simd.c

```diff
@@ -35,6 +37,78 @@ interp_v128_bitcast (gpointer res, gpointer v1)
 	*(v128_i1*)res = *(v128_i1*)v1;
 }
 
+// Vector2 AsVector2(Vector128<float> v1)
+static void
+interp_v128_to_v2 (gpointer res, gpointer v1)
+{
+	float *res_typed = (float*)res;
+	float *v1_typed = (float*)v1;
+
+	res_typed [0] = v1_typed [0];
+	res_typed [1] = v1_typed [1];
+}
+
+// Vector3 AsVector3(Vector128<float> v1)
+static void
+interp_v128_to_v3 (gpointer res, gpointer v1)
+{
+	float *res_typed = (float*)res;
+	float *v1_typed = (float*)v1;
+
+	res_typed [0] = v1_typed [0];
+	res_typed [1] = v1_typed [1];
+	res_typed [2] = v1_typed [2];
+}
+
+// Vector128<float> AsVector128(Vector2 v1)
+static void
+interp_v2_to_v128 (gpointer res, gpointer v1)
+{
+	float *res_typed = (float*)res;
+	float *v1_typed = (float*)v1;
+
+	res_typed [0] = v1_typed [0];
+	res_typed [1] = v1_typed [1];
+	res_typed [2] = 0;
+	res_typed [3] = 0;
+}
+
+// Vector3 AsVector3(Vector2 v1)
+static void
+interp_v2_to_v3 (gpointer res, gpointer v1)
+{
+	float *res_typed = (float*)res;
+	float *v1_typed = (float*)v1;
+
+	res_typed [0] = v1_typed [0];
+	res_typed [1] = v1_typed [1];
+	res_typed [2] = 0;
```

Comment:
There might be a problem here - if the v3 is in a stack local, it's 16 bytes wide and you might need to zero `[3]`. I'm not sure whether `res` can be a non-stack address though...

Reply:
I had code earlier that tried to always handle it as 16-bytes, but it didn't help.

Notably, I wouldn't expect the need to explicitly zero that space anyways as it would be considered padding space and should be ignored when loaded. Otherwise it would risk corrupting other operations, like Sum.